code/datums/radiation_wave.dm

```diff
@@ -112,8 +114,14 @@
 			))
 		if(!can_contaminate || blacklisted[thing.type])
 			continue
-		if(prob(contamination_chance)) // Only stronk rads get to have little baby rads
```

Comment:
You lost this prob call somewhere and you should be able to do all this well enough in one loop

Reply:
I didn't lose it, it's by design; all the contamination is always put somewhere on every rad pulse if it can be put anywhere, basically, it's WAY too weak without the prob

Reply:
I'd rather the contamination not always be put on the closest objects, perhaps increase the base chance and increase it the further out the wave goes. For example say in a simple case 50% base +10 per step means the first step would have half the contamination, second step 30%, third step 14% etc

Reply:
done, mostly